### PR TITLE
Fix missing etag when retrieving storage pool

### LINKED ISSUE #12431: Missing etag in storage-pool API response
# Required information

 * Distribution: Ubuntu Snap
 * Distribution version: 5.19

# Issue description

The `GET /1.0/storage-pools/big-pool` API response does not include an etag.

For doing updates from the UI it is useful to have an etag to avoid loosing changes that were done since requesting the pool data.

# Steps to reproduce

 1. Create a storage pool big-pool with driver zfs
 2. Query the pool details via `GET /1.0/storage-pools/big-pool`
 3. Check the response headers, no etag is included.

### PR DESCRIPTION
**Pull request by MusicDin**
Fixes #12431

### CODE REVIEW

**Comment by MusicDin** [comment]
@tomponline Should etag contain `pool.Driver()` or `pool.Driver().Info().Name`?

**Comment by tomponline** [comment]
But there is already an etag? what was the problem?

**Comment by MusicDin** [comment]
The function was passed instead of concrete value

File: lxd/storage_pools.go [hunk 1]
```diff
@@ -624,7 +624,7 @@ func storagePoolGet(d *Daemon, r *http.Request) response.Response {
 		poolAPI.Status = pool.LocalStatus()
 	}
 
-	etag := []any{pool.Name, pool.Driver, poolAPI.Config}
+	etag := []any{pool.Name(), pool.Driver().Info().Name, poolAPI.Config}
```
Comment (tomponline) [inline lxd/storage_pools.go:627]: I think we should use fields from poolAPI: Name, Config, Description

Comment (MusicDin) [reply lxd/storage_pools.go:627]: I made it equal as here in `storagePoolPut()`:

https://github.com/canonical/lxd/blob/7957bdf5541a4169c676f5ebb300b9ad857957c5/lxd/storage_pools.go#L714

Should I then also change it there?

Comment (tomponline) [reply lxd/storage_pools.go:627]: Ah ok do it as you propose then, but i think we should include Description in both.

**Review by tomponline — APPROVED** [review]

**Pull request merged**